src/test/java/spoon/test/pattern/TypePatternTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: (MIT OR CECILL-C)
+ *
+ * Copyright (C) 2006-2019 INRIA and contributors
+ *
+ * Spoon is available either under the terms of the MIT License (see LICENSE-MIT.txt) of the Cecill-C License (see LICENSE-CECILL-C.txt). You as the user are entitled to choose the terms under which to adopt Spoon.
+ */
+package spoon.test.pattern;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.CtModel;
+import spoon.reflect.code.BinaryOperatorKind;
+import spoon.reflect.code.CtBinaryOperator;
+import spoon.reflect.code.CtTypePattern;
+import spoon.reflect.visitor.filter.TypeFilter;
+import spoon.support.compiler.VirtualFile;
+import spoon.support.reflect.code.CtTypePatternImpl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TypePatternTest {
+
+	private static CtModel createModelFromString(String code) {
+		Launcher launcher = new Launcher();
+		launcher.getEnvironment().setComplianceLevel(16);
+		launcher.getEnvironment().setNoClasspath(true);
+		launcher.addInputResource(new VirtualFile(code));
+		return launcher.buildModel();
+	}
+
+	@Test
+	public void testWithInstanceOf() {
```

Comment:
Add an inline contract comment at the top of this test saying what it's supposed to test, it's not clear to me from the title alone.

```java
// contract: When X and Y then Z happens
```

Reply:
The test basically just checks if the model is built correctly (not sure if the BinaryOperatorKind even needs to be checked).

In general, it feels like one test isn't enough, I could probably add a test to check the pretty printed output or something like that...

*If* we're going to restrict accepted parent types, there would be more to test.

Reply:
That's all good, but there should still be a contract at the top of the test stating what it's supposed to verify :)